splink/linker.py

```diff
@@ -2008,6 +2008,51 @@ def count_num_comparisons_from_blocking_rule(
         res = self._execute_sql_pipeline().as_record_dict()[0]
         return res["count_of_pairwise_comparisons_generated"]
 
+    def cumulative_comparisons_from_blocking_rules_records(
+        self,
+        blocking_rules: str or list = None,
+    ):
+        """Output the number of comparisons generated by each successive blocking rule.
+
+        This is equivalent to the output size of df_predict and details how many
+        comparisons each of your individual blocking rules will contribute to the
+        total.
+
+        Args:
+            blocking_rules (str or list): The blocking rule(s) to compute comparisons
+                for. If null, the rules set out in your settings object will be used.
+
+        Examples:
+            >>> linker_settings = DuckDBLinker(df, settings)
+            >>> # Compute the cumulative number of comparisons generated by the rules
+            >>> # in your settings object.
+            >>> linker_settings.cumulative_comparisons_from_blocking_rules_records()
+            >>>
+            >>> # Generate total comparisons with custom blocking rules.
+            >>> blocking_rules = [
+            >>>    "l.surname = r.surname",
+            >>>    "l.first_name = r.first_name
+            >>>     and substr(l.dob,1,4) = substr(r.dob,1,4)"
+            >>> ]
+            >>>
+            >>> linker.cumulative_num_comparisons_from_blocking_rules_chart(
+            >>>     blocking_rules
+            >>>  )
+
+        Returns:
+            VegaLite: A VegaLite chart object. See altair.vegalite.v4.display.VegaLite.
+                The vegalite spec is available as a dictionary using the `spec`
+                attribute.
+        """
```

Comment:
This should be on the other method (i.e. the chart one) I think - this returns a record list